api/v1/helpers/verbs/utils.js

```diff
@@ -41,13 +41,11 @@ function updateInstance(o, puttableFields, toPut) {
       }
 
       o.set(key, nullish);
-
-      // take nullified fields out of changed fields
-      o.changed(key, false);
     } else {
       /*
-       * value may have changed. set changed to true to
-       * trigger checks in the model
+       * value may be set but not changed. set changed to true to
+       * trigger checks in the model. This is the only way to differentiate between
+       * PUT and PATCH in certain cases.
        */
       o.changed(key, true);
```

Comment:
I spent a lot of time trying to get it working without this line too, but wasn't able to. 

There are certain cases where a PUT and a PATCH can have the same "changed" values, but the behavior should be different. Setting changed to true here allows us to differentiate them. I don't think there's any way around this as long as we are using the same logic in the model update hooks for both PUT and PATCH.

Leaving in this line should be fine, since it's setting it to true. This just means it will always be updated in the db even if the value is the same.